README.md

```diff
@@ -0,0 +1,335 @@
+
+At this point in time, the configuration in the Alexa Developer Portal is about complete. The exception being the endpoint which doesn't exist yet.
+
+## Building a Lambda Function with Golang and MongoDB
+
+Alexa, for the most part should be able to direct requests, so now we need to create our backend to receive and process them. This is where Lambda, Go, and MongoDB come into play.
+
+Assuming Golang has been properly installed and configured, create a new project within your **$GOPATH** and within that project, create a **main.go** file. As boilerplate to get the ball rolling, this file should contain the following:
+
+```go
+package main
+
+func main() { }
+```
+
+With the boilerplate code added, now we can install the MongoDB Go driver. To do this, you could in theory do a `go get`, but the preferred approach as of now is to use the [dep](https://github.com/golang/dep) package management tool for Golang. To do this, after having installed the tool, execute the following:
+
+```bash
+dep init
+dep ensure -add "go.mongodb.org/mongo-driver/mongo"
+```
+
+We're using `dep` so that way the version of the driver that we're using in our project is version locked.
+
+In addition to the MongoDB Go driver, we're also going to need to get the [AWS Lambda SDK for Go](https://github.com/aws/aws-lambda-go) as well as an [unofficial SDK for Alexa](https://github.com/arienmalec/alexa-go), since no official SDK exists. To do this, we can execute:
+
+```bash
+dep ensure -add "github.com/arienmalec/alexa-go"
+dep ensure -add "github.com/aws/aws-lambda-go/lambda"
+```
+
+With the dependencies available to us, we can modify the project's **main.go** file. Open the file and add the following code:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Connection struct {
+	database *mongo.Database
+}
+
+func main() {
+	client, err := mongo.NewClient(options.Client().ApplyURI(os.Getenv("ATLAS_URI")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	err = client.Connect(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Disconnect(ctx)
+	database := client.Database("alexa")
+	connection := Connection{
+		database: database,
+	}
+}
+```
+
+In the `main` function we are creating a client using the connection string of our cluster. In this case, I'm using an environment variable on my computer that points to my MongoDB Atlas cluster. Feel free to configure that connection string however you feel the most confident.
+
+Before attempting to connect, we are setting a generous timeout of ten seconds. Upon connecting, we are getting a handle of an `alexa` database and storing it in a `Connection` data structure. If the `alexa` database doesn't exist, it will create one if we try to create data.
```

Comment:
`If the `alexa` database doesn't exist, it will create one if we try to create data.`

To me, this implies that the `Client.Database` call creates a new database object in MongoDB, but it actually doesn't do any I/O. It's creating a Go representation of a MongoDB database, and the first write operation will create the database in the server if necessary.